doctrine/registration_form.rst

```diff
@@ -178,207 +126,83 @@ the :doc:`Security Guide </security>`.
-        {
-            $builder
-                ->add('email', EmailType::class)
-                ->add('username', TextType::class)
-                ->add('plainPassword', RepeatedType::class, array(
-                    'type' => PasswordType::class,
-                    'first_options'  => array('label' => 'Password'),
-                    'second_options' => array('label' => 'Repeat Password'),
-                ))
-            ;
-        }
-
-        public function configureOptions(OptionsResolver $resolver)
-        {
-            $resolver->setDefaults(array(
-                'data_class' => User::class,
-            ));
-        }
-    }
-
-There are just three fields: ``email``, ``username`` and ``plainPassword``
-(repeated to confirm the entered password).
-
-.. tip::
-
-    To explore more things about the Form component, read the
-    :doc:`/forms` guide.
-
-Handling the Form Submission
-----------------------------
-
-Next, you need a controller to handle the form rendering and submission. If the
-form is submitted, the controller performs the validation and saves the data
-into the database::
-
-    // src/Controller/RegistrationController.php
     namespace App\Controller;
 
-    use App\Form\UserType;
     use App\Entity\User;
+    use App\Form\RegistrationFormType;
+    use App\Security\StubAuthenticator;
     use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
     use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
     use Symfony\Component\Routing\Annotation\Route;
     use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+    use Symfony\Component\Security\Guard\GuardAuthenticatorHandler;
 
     class RegistrationController extends AbstractController
     {
         /**
-         * @Route("/register", name="user_registration")
+         * @Route("/register", name="app_register")
          */
-        public function register(Request $request, UserPasswordEncoderInterface $passwordEncoder)
+        public function register(Request $request, UserPasswordEncoderInterface $passwordEncoder): Response
         {
-            // 1) build the form
-            $user = new User();
-            $form = $this->createForm(UserType::class, $user);
-
-            // 2) handle the submit (will only happen on POST)
+            $form = $this->createForm(RegistrationFormType::class);
             $form->handleRequest($request);
+
             if ($form->isSubmitted() && $form->isValid()) {
+                /** @var User */
+                $user = $form->getData();
 
-                // 3) Encode the password (you could also do this via Doctrine listener)
-                $password = $passwordEncoder->encodePassword($user, $user->getPlainPassword());
-                $user->setPassword($password);
+                // encode the plain password
+                $user->setPassword(
+                    $passwordEncoder->encodePassword(
```

Comment:
In this comment https://github.com/symfony/maker-bundle/pull/333#issuecomment-449400566  it was suggested to do the password transform in a form listener. I don't have a strong opinion about it. Can that "listener" be defined in a simple method of the registration form? I'd prefer to not create a new class in a different form namespace to implement this simple feature. Thanks!

Reply:
The listener can be inlined in the form, yest

Reply:
This is not my preferred way of doing it - I dislike putting any of my logic into the form system if I can avoid it - it adds complexity, and I don't see the benefit.

Reply:
I can understand both positions ... let me try to give another argument about moving this out of the controller:

* Right now, the controller needs to encode the password because the form gave it the "wrong" password. If I were the controller, I'd think: *"damn, this Form is an idiot; instead of giving me the right data, it gives me some "unfinished" information and I need to fix it myself before doing my job."*
* If the logic was complex, I understand that it has to be done in the controller (or a service called from the controller). The Form doesn't have to know anything about complex logic.
* However, encoding the password to me looks like Form's responsibility, because the Form must give the controller the right data. The fact that the user inputs a non-encoded password while the controller needs the encoded one is like an internal detail that the Form could implement to simplify everything else.

In any case, all this is true to me if this password encoding is done in a simple method inside the form. No more classes and complications, please.